openedx/core/djangoapps/course_apps/tests/test_wiki_app.py

```diff
@@ -0,0 +1,57 @@
+"""
+Tests for wiki course app.
+"""
+
+from common.djangoapps.student.tests.factories import AdminFactory, UserFactory
+from lms.djangoapps.course_wiki.plugins.course_app import WikiCourseApp
+from openedx.core.djangolib.testing.utils import skip_unless_cms
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+
+@skip_unless_cms
+class WikiCourseAppTestCase(ModuleStoreTestCase):
+    """Test cases for Wiki CourseApp."""
+
+    def setUp(self):
+        super().setUp()
+        self.course = CourseFactory.create()
+        self.instructor = AdminFactory.create()
+        self.user = UserFactory()
+
+    def get_wiki_tab(self, course_key):
+        """
+        Reload the course and fetch the wiki tab if present.
+        """
+        course = self.store.get_course(course_key)
+        return next((tab for tab in course.tabs if tab.type == 'wiki'), None)
+
+    def test_app_disabled_by_default(self):
+        """
+        Test that the wiki tab is disabled by default.
+        """
+        assert not WikiCourseApp.is_enabled(self.course.id)
+
+    def test_app_enabling(self):
+        """
+        Test that enabling and disable the app enabled/disables the tab.
```

Comment:
suggestion: might be better to have separate tests for **enabling** and **disabling** the wiki tab! Just in case we have more in-depth test cases in the future!

Reply:
I'm not entirely sure how make such a separation. The tabs will default to disabled. So in order to test disabling it, you need to enable it first. If you're enabling it, you should test that it's actually enabled, before disabling it otherwise the disable test will do nothing.